lib/util/http/gitea.ts

```diff
@@ -4,7 +4,7 @@ import { Http, HttpOptions, HttpResponse, InternalHttpOptions } from '.';
 
 let baseUrl: string;
 export const setBaseUrl = (newBaseUrl: string): void => {
-  baseUrl = newBaseUrl.replace(/\/*$/, '/'); // TODO #12070 #12071
+  baseUrl = newBaseUrl.replace(/\/*$/, '/'); // TODO #12070 #12071 adds slash at front when re2 is used
```

Comment:
`ensureTrailingSlash`

Reply:
This regex -> `/\/*$/` and the `ensureTrailingSlash` are slighlty different.
ensureTrailingSlash only matches the first `/` whereas `/\/*$/` matches all the slashes.
Both will have different result for this test case: `foo//////`

Reply:
@rarkins Should we update `ensureTrailingSlash` to match multiple trailing slashes? I think that would be a config error, so we should simply ignore multiple trailing slashes.